---
title: "animalcules"
author:
- name: Yue Zhao
  affiliation:
  - &1 Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA
  - &2 Bioinformatics Program, Boston University, Boston, MA
- name: Anthony Federico
  affiliation:
  - *1
  - *2
- name: Evan Johnson
  affiliation:
  - *1
  - *2
date: '`r format(Sys.Date(), "%B %e, %Y")`'
abstract: 
     animalcules is an R package for utilizing up-to-date data analytics,
     visualization methods, and machine learning models to provide users 
     an easy-to-use interactive microbiome analysis framework. It can be 
     used as a standalone software package or users can explore their data 
     with the accompanying interactive R Shiny application. Traditional 
     microbiome analysis such as alpha/beta diversity and differential 
     abundance analysis are enhanced, while new methods like biomarker 
     identification are introduced by animalcules. Powerful interactive 
     and dynamic figures generated by animalcules enable users to 
     understand their data better and discover new insights. 
package: animalcules
output:
    BiocStyle::html_document
vignette: >
    %\VignetteIndexEntry{animalcules}
    %\VignetteEncoding{UTF-8}
    %\VignetteEngine{knitr::rmarkdown}
editor_options:
    chunk_output_type: console
---


# Knit options

```{r}
knitr::opts_chunk$set(message = FALSE)
```

# Packages

```{r}
library(magrittr)
library(tibble)
library(MultiAssayExperiment)
library(SummarizedExperiment)
library(S4Vectors)
library(usethis)
library(dplyr)
library(plotly)
library(tsne)
library(reshape2)
library(caret)
library(vegan)
library(ape)
library(DESeq2)
library(ggplot2)
library(forcats)
library(grid)
library(lattice)
library(plotROC)
library(animalcules)

```


# Load toy dataset

This toy dataset contains 50 simulated samples. Throughout the vignette, we will use the data 
structure MultiAssayExperiment (MAE) as the input to all functions.

```{r}
data_dir = system.file("extdata/MAE.rds", package = "animalcules")
MAE = readRDS(data_dir)
```


# run shiny app

To run the shiny app, just call: run_animalcules(). The rest of the vignette is for the command line
version of animalcules. 

```{r}
# run_animalcules()
```



# Summary and Categorize

## Summary Plot Top

```{r}
p <- filter_summary_top(MAE,
                        samples_discard = c("subject_2", "subject_4"),
                        filter_type = "By Metadata",
                        sample_condition = "AGE")
p
```

## Summary Plot Bottom

```{r}
p <- filter_summary_bottom(MAE,
                           samples_discard = c("subject_2", "subject_4"),
                           filter_type = "By Metadata",
                           sample_condition = "SEX")
p
```

## Categorize

```{r}
microbe <- MAE[['MicrobeGenetics']]
samples <- as.data.frame(colData(microbe))
result <- filter_categorize(samples,
                            sample_condition = "AGE",
                            new_label="AGE_GROUP",
                            bin_breaks=c(0,55,75,100),
                            bin_labels=c('Young','Adult',"Elderly"))
result$sam_table
result$plot.unbinned
result$plot.binned
```

# Visualization

## Relative Abundance Stacked Bar Plot



```{r}
p <- relabu_barplot(MAE,
                    tax_level="family",
                    order_organisms=c('Retroviridae'),
                    sort_by="organisms",
                    sample_conditions=c('SEX', 'AGE'),
                    show_legend=TRUE)
p
```

## Relative Abundance Heatmap

```{r}
p <- relabu_heatmap(MAE,
                   tax_level="genus",
                   sort_by="conditions",
                   sample_conditions=c("SEX", "AGE"))
p
```

## Relative Abundance Boxplot

```{r}
p <- relabu_boxplot(MAE,
                    tax_level="genus",
                    organisms=c("Escherichia", "Actinomyces"),
                    condition="SEX",
                    datatype="logcpm")
p
```

# Diversity

## alpha diversity boxplot

Alpha diversity, which describes the richness and evenness of sample microbial community, is a vital indicator in the microbiome analysis. animalcules provides the interactive boxplot comparison of alpha diversity between selected groups of samples. Both taxonomy levels and alpha diversity metrics (Shannon, Gini Simpson, Inverse Simpson) can be changed. Users can also conduct alpha diversity statistical tests including Wilcoxon rank sum test, T test and Kruskal-Wallis test.  

Plot the alpha diversity boxplot between the levels in selected condition.

```{r}
alpha_div_boxplot(MAE = MAE,
                  tax_level = "genus",
                  condition = "DISEASE",
                  alpha_metric = "shannon")
```

## alpha diversity statistical test

Conduct statistical test on the alpha diversity between the levels in selected condition.

```{r}
do_alpha_div_test(MAE = MAE,
                  tax_level = "genus",
                  condition = "DISEASE",
                  alpha_metric = "shannon",
                  alpha_stat = "T-test")
```

## beta diversity heatmap

On the other hand, by defining distances between each sample, beta diversity is another key metric to look at. Users can plot the beta diversity heatmap by selecting different beta diversity dissimilarity metrics including Bray-Curtis and Jaccard. Users can also conduct beta diversity statistical testing between groups including PERMANOVA, Wilcoxon rank sum test and Kruskal-Wallis.

Plot the beta diversity heatmap with selected condition.

```{r}
diversity_beta_heatmap(MAE = MAE, 
                       tax_level = 'genus', 
                       input_beta_method = "bray",
                       input_bdhm_select_conditions = 'DISEASE',
                       input_bdhm_sort_by = 'condition')
```

## beta boxplot

Plot the beta diversity boxplot within and between conditions.

```{r}
diversity_beta_boxplot(MAE = MAE, 
                       tax_level = 'genus', 
                       input_beta_method = "bray",
                       input_select_beta_condition = 'DISEASE')
```

## beta diversity stat

Conduct statistical test on the beta diversity between the levels in selected condition.

```{r}
diversity_beta_test(MAE = MAE, 
                    tax_level = 'genus',
                    input_beta_method = "bray",
                    input_select_beta_condition =  'DISEASE',
                    input_select_beta_stat_method = 'PERMANOVA',
                    input_num_permutation_permanova = 999)
```

# Server_05_dimred

# PCA
```{r}
result <- dimred_pca(MAE,
                     tax_level="genus",
                     color="AGE",
                     shape="DISEASE",
                     pcx=1,
                     pcy=2,
                     datatype="logcpm")
#result$plot
result$table
```

# PCoA
```{r}
result <- dimred_pcoa(MAE,
                      tax_level="genus",
                      color="AGE",
                      shape="DISEASE",
                      axx=1,
                      axy=2,
                      method="bray")
#result$plot
result$table
```

# t-SNE
```{r}
p <- dimred_tsne(MAE,
                 tax_level="phylum",
                 color="AGE",
                 shape="GROUP",
                 k="3D",
                 initial_dims=30,
                 perplexity=10,
                 datatype="logcpm")
p
```

# Server_06_differential_analysis

Here in animalcules, we provide a DESeq2-based differential abundance analysis. Users can choose the target variable, covariate variable, taxonomy level, minimum count cut-off, and an adjusted p-value threshold. The analysis report will output not only the adjusted p-value and log2-fold-change of the microbes, but also the percentage, prevalence, and the group size-adjusted fold change.

```{r}
p <- differential_abundance(MAE,
                            tax_level="genus",
                            input_da_condition=c("DISEASE"),
                            input_da_condition_covariate=c("SEX"),
                            min_num_filter = 5,
                            input_da_padj_cutoff = 0.5)
p
```

# Server_07_biomarker


One unique feature of animalcules is the biomarker identification module built on machine learning models. Users can choose one classification model from logistic regression, gradient boosting machine, or random forest to identify a microbes biomarker list. The feature importance score for each microbe will be provided. To evaluate the biomarker performance, the ROC plot and the AUC value using cross-validation outputs are shown to users. 

## Train biomarker
```{r}
p <- find_biomarker(MAE,
                    tax_level="genus",
                    input_select_target_biomarker=c("SEX"),
                    nfolds = 3,
                    nrepeats = 3,
                    seed = 99,
                    percent_top_biomarker = 0.2,
                    model_name = "logistic regression")
# biomarker
p$biomarker

# importance plot
p$importance_plot

# ROC plot
p$roc_plot
```



# Toy Dataset

animalcules.rds is an .rds file that contains three tables: count_table, gene_expression_table
and metadata_table. Users could build their own MAE object using the code below:

```{r}
# data_raw <-
#     base::system.file("extdata/animalcules.rds", package = "animalcules") %>%
#     base::readRDS()
# 
# se_mgx <-
#     magrittr::use_series(data_raw, count_table) %>%
#     base::data.matrix() %>%
#     S4Vectors::SimpleList() %>%
#     magrittr::set_names("MGX")
# 
# se_ge <-
#     magrittr::use_series(data_raw, gene_expression_table) %>%
#     base::data.matrix() %>%
#     S4Vectors::SimpleList() %>%
#     magrittr::set_names("GeneExpression")
# 
# se_colData <-
#     magrittr::use_series(data_raw, metadata_table) %>%
#     S4Vectors::DataFrame()
# 
# se_rowData <-
#     magrittr::use_series(data_raw, tax_table) %>%
#     base::data.frame() %>%
#     dplyr::mutate_all(as.character) %>%
#     dplyr::select(superkingdom, phylum, class, order, family, genus) %>%
#     S4Vectors::DataFrame()
# 
# microbe_se <-
#     SummarizedExperiment::SummarizedExperiment(assays = se_mgx,
#                                                colData = se_colData,
#                                                rowData = se_rowData)
# 
# host_se <-
#     SummarizedExperiment::SummarizedExperiment(assays = se_ge,
#                                                colData = se_colData)
# 
# mae_experiments <-
#     S4Vectors::SimpleList(MicrobeGenetics = microbe_se, HostGenetics = host_se)
# 
# MAE <-
#     MultiAssayExperiment::MultiAssayExperiment(experiments = mae_experiments,
#                                                colData = se_colData)

## usethis::use_data(MAE)

# saveRDS(MAE, "extdata/MAE.rds")
```



# Session Info

```{r}
sessionInfo()
```
