<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interactive microbiome analysis toolkit • animalcules</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Interactive microbiome analysis toolkit">
<meta property="og:description" content="
    animalcules is an R package for utilizing up-to-date data analytics,
    visualization methods, and machine learning models to provide users
    an easy-to-use interactive microbiome analysis framework. It can be
    used as a standalone software package or users can explore their data
    with the accompanying interactive R Shiny application. Traditional
    microbiome analysis such as alpha/beta diversity and differential
    abundance analysis are enhanced, while new methods like biomarker
    identification are introduced by animalcules. Powerful interactive
    and dynamic figures generated by animalcules enable users to
    understand their data better and discover new insights.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">animalcules</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.71</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/animalcules.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/animalcules">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">

<p align="center"><img width="80%" src="https://github.com/compbiomed/animalcules/blob/master/inst/shiny/www/animalcules.jpg"></p>
<p>                                                                              </p>
<p><a href="https://doi.org/doi:10.18129/B9.bioc.animalcules"><img src="https://img.shields.io/badge/bioconductor-3.9-3a6378.svg"></a> <a href="https://bioconductor.org/checkResults/3.9/bioc-LATEST/animalcules/"><img src="https://img.shields.io/badge/platforms-linux%20%7C%20osx%20%7C%20win-2a89a1.svg"></a> <a href="https://github.com/compbiomed/animalcules/commits/master"><img src="https://img.shields.io/github/last-commit/compbiomed/animalcules.svg"></a> <a href="https://www.tidyverse.org/lifecycle/#maturing"><img src="https://img.shields.io/badge/lifecycle-maturing-blue.svg"></a> <a href="https://bioconductor.org/checkResults/devel/bioc-LATEST/animalcules/"><img src="https://bioconductor.org/shields/build/devel/bioc/animalcules.svg"></a></p>
<div id="table-of-contents" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#table-of-contents" class="anchor"></a>Table of Contents</h1></div>
<ul>
<li>
<a href="#table-of-contents">Table of Contents</a>
<ul>
<li><a href="#what-does-animalcules-do">What does animalcules do?</a></li>
<li><a href="#installation">Installation</a></li>
<li><a href="#run-animalcules">Run animalcules</a></li>
<li><a href="#docs-for-command-line">Docs for command line</a></li>
<li>
<a href="#docs-for-shiny-app">Docs for shiny app</a>
<ul>
<li>
<a href="#tab-1-upload">Tab 1: Upload</a>
<ul>
<li><a href="#example-data">Example Data</a></li>
<li><a href="#count-file">Count File</a></li>
<li><a href="#pathoscope-file">Pathoscope File</a></li>
<li><a href="#animalcules-id-file">animalcules-id file</a></li>
<li><a href="#animalcules-file">animalcules file</a></li>
</ul>
</li>
<li>
<a href="#tab-2-summary-and-filter">Tab 2: Summary and Filter</a>
<ul>
<li><a href="#summary-plots">Summary Plots</a></li>
<li><a href="#categorize">Categorize</a></li>
</ul>
</li>
<li>
<a href="#tab-3-abundance">Tab 3: Abundance</a>
<ul>
<li><a href="#relative-abundance-stacked-bar-plot">Relative Abundance Stacked Bar Plot</a></li>
<li><a href="#relative-abundance-heatmap">Relative Abundance Heatmap</a></li>
<li><a href="#relative-abundance-boxplot">Relative Abundance Boxplot</a></li>
</ul>
</li>
<li>
<a href="#tab-4-diversity">Tab 4: Diversity</a>
<ul>
<li><a href="#alpha-diversity">Alpha Diversity</a></li>
<li><a href="#beta-diversity-heatmap">Beta Diversity Heatmap</a></li>
<li><a href="#beta-diversity-statistical-test">Beta Diversity Statistical Test</a></li>
</ul>
</li>
<li>
<a href="#tab-5-dimension-reduction">Tab 5: Dimension Reduction</a>
<ul>
<li><a href="#pca">PCA</a></li>
<li><a href="#pcoa">PCoA</a></li>
<li><a href="#t-sne">t-SNE</a></li>
</ul>
</li>
<li>
<a href="#tab-6-differential-analysis">Tab 6: Differential Analysis</a>
<ul>
<li><a href="#run-differential-analysis">Run Differential Analysis</a></li>
</ul>
</li>
<li>
<a href="#tab-7-biomarker">Tab 7: Biomarker</a>
<ul>
<li><a href="#run-biomarker">Run Biomarker</a></li>
<li><a href="#importance-plot">Importance Plot</a></li>
<li><a href="#cv-roc-plot">CV ROC Plot</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div id="what-does-animalcules-do" class="section level2">
<h2 class="hasAnchor">
<a href="#what-does-animalcules-do" class="anchor"></a>What does animalcules do?</h2>
<p><strong>R shiny app for interactive microbiome analysis</strong></p>
<p>animalcules is an R package for utilizing up-to-date data analytics, visualization methods, and machine learning models to provide users an easy-to-use interactive microbiome analysis framework. It can be used as a standalone software package or users can explore their data with the accompanying interactive R Shiny application.</p>
<p>Traditional microbiome analysis such as alpha/beta diversity and differential abundance analysis are enhanced, while new methods like biomarker identification are introduced by animalcules. Powerful interactive and dynamic figures generated by animalcules enable users to understand their data better and discover new insights.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<ul>
<li>Install from Bioconductor(not yet!)</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ns-load">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly=</span><span class="ot">TRUE</span>))
  <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"BiocManager"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"animalcules"</span>)</code></pre></div>
<ul>
<li>Install from github</li>
</ul>
<p>Note: Install devtools first if you haven’t installed it yet.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages("devtools")</span>
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"compbiomed/animalcules"</span>)</code></pre></div>
</div>
<div id="run-animalcules" class="section level2">
<h2 class="hasAnchor">
<a href="#run-animalcules" class="anchor"></a>Run animalcules</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(animalcules)
<span class="kw"><a href="reference/run_animalcules.html">run_animalcules</a></span>()</code></pre></div>
</div>
<div id="docs-for-command-line-not-yet" class="section level2">
<h2 class="hasAnchor">
<a href="#docs-for-command-line-not-yet" class="anchor"></a>Docs for command line (Not yet!)</h2>
<p>Please check the documents for command line based animalcules in the <a href="https://bioconductor.org/packages/release/bioc/html/animalcules.html">animalcules vignette</a> at Bioconductor.</p>
</div>
<div id="docs-for-shiny-app" class="section level2">
<h2 class="hasAnchor">
<a href="#docs-for-shiny-app" class="anchor"></a>Docs for shiny app</h2>
<div id="tab-1-upload" class="section level3">
<h3 class="hasAnchor">
<a href="#tab-1-upload" class="anchor"></a>Tab 1: Upload</h3>
<div id="example-data" class="section level5">
<h5 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example Data</h5>
<p>There are two example datasets available in animalcules: a synthetic toy dataset (already loaded) containing 50 samples and a real TB dataset containing 30 samples. Users could use any dataset to try all functions and features in animalcules.</p>
<p>Instructions:</p>
<ul>
<li>Use the pulldown menu to select the example dataset.</li>
<li>Click the “Upload” button.</li>
</ul>
<p>Running time: &lt; 0.5s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_030.png" width="400"></p>
</div>
<div id="count-file" class="section level5">
<h5 class="hasAnchor">
<a href="#count-file" class="anchor"></a>Count File</h5>
<p>Both 16s and metagenomics generated data are supported here. The three required files are:</p>
<ul>
<li>Counts file: each row is a species/OTU, each column is a sample name.</li>
<li>Taxonomy file: each row is a species/OTU, each column is a taxonomy level.</li>
<li>Annotation file: each row is a sample name, each column is a variable/feature name.</li>
</ul>
<p>Make sure:</p>
<ul>
<li>The sample names in the counts file agree with the one in annotation file.</li>
<li>The microbe names in the counts file agree with the ones in the taxonomy file.</li>
<li>All three files have the same file separator / same file type (e.g. csv).</li>
</ul>
<p>Instructions:</p>
<ul>
<li>Click the “Browse…” button to upload required three files.</li>
<li>Input which column in the annotation file is the sample name (default is 1).</li>
<li>Check if the annotation file has header (default is TRUE).</li>
<li>Select the separator (default is comma AKA .csv file).</li>
<li>Click the button “Upload”.</li>
</ul>
<p>Note: Each uploaded table will show up in the right panel.</p>
<p>Running time: &lt; 1s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_031.png" width="300"></p>
</div>
<div id="pathoscope-file" class="section level5">
<h5 class="hasAnchor">
<a href="#pathoscope-file" class="anchor"></a>Pathoscope File</h5>
<p>To analyze pathoscope outputs, users need to upload pathoscope reports (use browser for multiple reports upload), as well as an annotation file containing metadata for each sample. Note that the sample name in the annotation file must match the non-suffix part of the pathoscope file name. For example, one pathoscope report filename is: “sample_011-sam-report.tsv”, then the corresponding sample name in the annotation file must be: “sample_011”.</p>
<p>Instructions:</p>
<ul>
<li>Click the “Browse…” button to upload required files.</li>
<li>Specify the pathoscope report file suffix (default is -sam-report.tsv).</li>
<li>Input which column in the annotation file is the sample name (default is 1).</li>
<li>Check if the annotation file has header (default is TRUE).</li>
<li>Select the separator (default is Tab AKA .tsv file).</li>
<li>Click the button “Upload”.</li>
</ul>
<p>Also, make sure to provide the correct column number for sample name in the annotation file, as well as the annotation file separator (tab, comma or semicolon).</p>
<p>Note: One example pathoscope eport table and the annotation table will show up in the right panel.</p>
<p>Running time:</p>
<ul>
<li>Test dataset A with 30 samples: 10.8s</li>
<li>Test dataset B with 587 samples: 22.1s</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_032.png" width="300"></p>
</div>
<div id="animalcules-id-file" class="section level5">
<h5 class="hasAnchor">
<a href="#animalcules-id-file" class="anchor"></a>animalcules-id file</h5>
<p>animalcules-id is a separate R pipeline that generates pathoscope-like outputs from fastq files. The required input is the animalcules-id generated .rds file. Here users could choose either EM count assay or Best hit assay.</p>
<p>Running time: &lt; 1s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_033.png" width="400"></p>
</div>
<div id="animalcules-file" class="section level5">
<h5 class="hasAnchor">
<a href="#animalcules-file" class="anchor"></a>animalcules file</h5>
<p>In animalcules, users could choose to save their dataset to a .rds file in the Tab 2 (Summary and Filter). Later, users could load this saved dataset by uploading the .rds file to animalcules easily via this animalcules file upload option.</p>
<p>Running time: &lt; 0.5s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_034.png" width="400"></p>
</div>
</div>
<div id="tab-2-summary-and-filter" class="section level3">
<h3 class="hasAnchor">
<a href="#tab-2-summary-and-filter" class="anchor"></a>Tab 2: Summary and Filter</h3>
<div id="summary-plots" class="section level5">
<h5 class="hasAnchor">
<a href="#summary-plots" class="anchor"></a>Summary Plots</h5>
<p>One type of summarization plot returns either a box plot or pie chart for continous or categorical data respectively while another type returns returns either a density plot or bar plot for continous or categorical data respectively.</p>
<p>Instructions:</p>
<ul>
<li>Filter by either metadata or microbe information</li>
<li>Select a condition to filter by</li>
<li>Choose a cutoff point or range of acceptable values</li>
<li>(Optional) Discard samples or organisms</li>
<li>Click filter to filter the datasaet</li>
<li>Click reset to remove all previous applied filters</li>
<li>Click download to export the modified animacules file for later use</li>
</ul>
<p>Running time: &lt; 1s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_042.png"></p>
</div>
<div id="categorize" class="section level5">
<h5 class="hasAnchor">
<a href="#categorize" class="anchor"></a>Categorize</h5>
<p>It is often necessary to bin continuous data into categories when performing analyses that require categorical input. To help ease this process, users can automatically categorize categorical data and provide custom bin breaks and labels in doing so.</p>
<p>Instructions:</p>
<ul>
<li>Select a continous variable to categorize</li>
<li>(Optional) Choose the number of bins</li>
<li>(Optional) Choose a new variable name</li>
<li>(Optional) Set custom breaks to define the new bins</li>
<li>(Optional) Add custom labels to each of the bins</li>
<li>Click “Create Bins”</li>
</ul>
<p>Running time: &lt; 1s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_043.png"></p>
</div>
</div>
<div id="tab-3-abundance" class="section level3">
<h3 class="hasAnchor">
<a href="#tab-3-abundance" class="anchor"></a>Tab 3: Abundance</h3>
<div id="relative-abundance-stacked-bar-plot" class="section level5">
<h5 class="hasAnchor">
<a href="#relative-abundance-stacked-bar-plot" class="anchor"></a>Relative Abundance Stacked Bar Plot</h5>
<p>The stacked bar plots are used to visualize the relative abundance of microbes at a given taxonomical level in each sample represented as a single bar.</p>
<p>Instructions:</p>
<ul>
<li>Select taxonomy level in the menu (default is genus)</li>
<li>(Optional) Add a color bar for one of more sample conditions</li>
<li>(Optional) Group samples and aggregate abundances by a condition</li>
<li>(Optional) Sort samples by conditions or the presence of specific organisms</li>
<li>(Optional) Isolate or discard samples</li>
<li>(Optional) Hide or show legend</li>
<li>(Optional) Adjust plot height</li>
<li>Click the button “Plot”</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_047.png"></p>
</div>
<div id="relative-abundance-heatmap" class="section level5">
<h5 class="hasAnchor">
<a href="#relative-abundance-heatmap" class="anchor"></a>Relative Abundance Heatmap</h5>
<p>The heatmap represents a sample by organisms matrix that can be visualized at different taxonomic levels.</p>
<p>Instructions:</p>
<ul>
<li>Select taxonomy level in the menu (default is genus)</li>
<li>(Optional) Add a color bar for one of more sample conditions</li>
<li>(Optional) Sort samples by conditions or the presence of specific organisms</li>
<li>(Optional) Isolate or discard samples</li>
<li>(Optional) Use raw counts or log(CPM)</li>
<li>(Optional) Adjust plot height</li>
<li>Click the button “Plot”</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_048.png"></p>
</div>
<div id="relative-abundance-boxplot" class="section level5">
<h5 class="hasAnchor">
<a href="#relative-abundance-boxplot" class="anchor"></a>Relative Abundance Boxplot</h5>
<p>The boxplot visualization allows users to compare the abundance of one or more organisms at multiple taxonomical levels between categorical attributes.</p>
<p>Instructions:</p>
<ul>
<li>Select one or more taxonomy levels in the menu (default is genus)</li>
<li>For each taxonomy level select one or more organisms to compare</li>
<li>(Optional) Separate plots when viewing multiple organisms on the same taxonomy level</li>
<li>Select a condition to compare organisms across</li>
<li>Select a datatype to use</li>
<li>Click the button “Plot”</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_049.png"></p>
</div>
</div>
<div id="tab-4-diversity" class="section level3">
<h3 class="hasAnchor">
<a href="#tab-4-diversity" class="anchor"></a>Tab 4: Diversity</h3>
<div id="alpha-diversity" class="section level5">
<h5 class="hasAnchor">
<a href="#alpha-diversity" class="anchor"></a>Alpha Diversity</h5>
<p>Users could visualize the alpha diversity boxplot comparing different levels within a selected condition by selecting taxonomy levels and target condition. In Advanced options, users could choose different alpha diversity metrics including Shannon, Inverse Simpson and Gini Simpson.</p>
<p>The statistical test will be performed between levels in the selected target condition. Multiple statistical test methods will be applied automatically, including T-test, Kruskal-Wallis rank sum test, and Wilcoxon rank sum test. Test p-values will be showed in the result table.</p>
<p>Instructions:</p>
<ul>
<li>Select taxonomy level in the menu (default is genus).</li>
<li>Select the target variable for alpha diversity comparison.</li>
<li>(Optional) Select alpha diversity metrics.</li>
<li>Click the button “Run”</li>
</ul>
<p>Running time: &lt; 1s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_035.png"></p>
</div>
<div id="beta-diversity-heatmap" class="section level5">
<h5 class="hasAnchor">
<a href="#beta-diversity-heatmap" class="anchor"></a>Beta Diversity Heatmap</h5>
<p>Users could visualize the beta diversity by selecting taxonomy levels, sample color condition. Samples could also be sorted by the selected condition. In Advanced options, users could choose different beta diversity metrics including Bray-curtis or Jaccard.</p>
<p>Instructions:</p>
<ul>
<li>Select taxonomy level in the menu (default is genus).</li>
<li>Select the target variable for beta diversity comparison.</li>
<li>Select if samples will be sorted by the selected variable (default: no sorting).</li>
<li>(Optional) Select beta diversity metrics.</li>
<li>Click the button “Plot Heatmap”</li>
</ul>
<p>Running time: &lt; 1s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_036.png"></p>
</div>
<div id="beta-diversity-statistical-test" class="section level5">
<h5 class="hasAnchor">
<a href="#beta-diversity-statistical-test" class="anchor"></a>Beta Diversity Statistical Test</h5>
<p>To check if there is significant beta diversity difference existing between two communities, users could also perform statistical test like PERMANOVA test, Kruskal-Wallis rank sum test, and Wilcoxon rank sum test. Note that the left panel options apply to both beta diversity heatmap and statistical test. Test p-value will be showed in the “Pr(&gt;F)” column from the result table.</p>
<p>Instructions:</p>
<ul>
<li>Click the tab “Boxplot” in the right panel next to the tab “Heatmap”</li>
<li>Select the test type (default is PERMANOVA).</li>
<li>Select the target variable.</li>
<li>(Optional) Select the number of permutations if doing PERMANOVA.</li>
<li>Click the button “Run”</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_037.png"></p>
</div>
</div>
<div id="tab-5-dimension-reduction" class="section level3">
<h3 class="hasAnchor">
<a href="#tab-5-dimension-reduction" class="anchor"></a>Tab 5: Dimension Reduction</h3>
<div id="pca" class="section level5">
<h5 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h5>
<p>A wrapper for conduction 2D and 3D Principal Component Analysis.</p>
<p>Instructions:</p>
<ul>
<li>Select one or more taxonomy levels in the menu (default is genus)</li>
<li>Color data points by a condition</li>
<li>Select variable on x-axis (default is PC1)</li>
<li>Select variable on y-axis (default is PC1)</li>
<li>(Optional) Select variable on z-axis for 3D plotting</li>
<li>(Optional) Shape data points by a condition</li>
<li>Select a datatype to use</li>
<li>Click the button “Plot” to plot</li>
<li>Click the button “Table” to see PCA data</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_044.png"></p>
</div>
<div id="pcoa" class="section level5">
<h5 class="hasAnchor">
<a href="#pcoa" class="anchor"></a>PCoA</h5>
<p>A wrapper for conduction 2D and 3D Principal Coordinate Analysis.</p>
<p>Instructions:</p>
<ul>
<li>Select one or more taxonomy levels in the menu (default is genus)</li>
<li>Color data points by a condition</li>
<li>Select variable on x-axis (default is PC1)</li>
<li>Select variable on y-axis (default is PC1)</li>
<li>(Optional) Select variable on z-axis for 3D plotting</li>
<li>(Optional) Shape data points by a condition</li>
<li>Select a distance metric to use</li>
<li>Click the button “Plot” to plot</li>
<li>Click the button “Table” to see PCoA data</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_045.png"></p>
</div>
<div id="t-sne" class="section level5">
<h5 class="hasAnchor">
<a href="#t-sne" class="anchor"></a>t-SNE</h5>
<p>A wrapper for conduction 2D and 3D t-distributed stochastic neighbor embedding.</p>
<p>Instructions:</p>
<ul>
<li>Select one or more taxonomy levels in the menu (default is genus)</li>
<li>Color data points by a condition</li>
<li>(Optional) Select final dimensions (2D or 3D)</li>
<li>(Optional) Shape data points by a condition</li>
<li>(Optional) Select t-SNE perplexity</li>
<li>(Optional) Select t-SNE initial dimensions</li>
<li>Select a datatype to use</li>
<li>Click the button “Plot” to plot</li>
<li>Select “Use Cached Data” to change color/shape of data points without re-running t-SNE</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_046.png"></p>
</div>
</div>
<div id="tab-6-differential-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#tab-6-differential-analysis" class="anchor"></a>Tab 6: Differential Analysis</h3>
<div id="run-differential-analysis" class="section level5">
<h5 class="hasAnchor">
<a href="#run-differential-analysis" class="anchor"></a>Run Differential Analysis</h5>
<p>To identify key microbes for a specific variable, users need to specify the taxonomy level and target variable (condition). In the Advanced Options, users could also add covariates to the linear model, add a minimum count cut-off (all features with average read number less than this cut-off will be filtered), and a adjusted p-value cut-off.</p>
<p>After click the “Run” button, users would see a differential abundance analysis output table on the right-hand side. For categorical variables in DESeq2 method, we show the feature name, adjusted p-value, original p-value, log2 fold change, number of samples for each class, feature prevalance, and group size adjusted fold change. For numeric variables in DESeq2 method, number of samples for each class and group size adjusted fold change won’t show up. For limma method, only adjusted p-value and original p-value will show up.</p>
<p>Instructions:</p>
<ul>
<li>Select either tab “DESeq2” or “limma” for the analysis (default is DESeq2).</li>
<li>Select taxonomy level in the menu (default is genus).</li>
<li>Select the target variable for differential abundance analysis.</li>
<li>(Optional) Select covariates.</li>
<li>(Optional) Select minimum total count cuf-off for microbes (default 500).</li>
<li>(Optional) Select adjusted p-value threshold (default 0.8).</li>
<li>Click the button “Run”</li>
</ul>
<p>Running time:</p>
<ul>
<li>Test dataset with 30 samples and 427 microbes: 2.48s</li>
<li>Test dataset with 587 samples and 203 microbes: 46.59s</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_038.png"></p>
</div>
</div>
<div id="tab-7-biomarker" class="section level3">
<h3 class="hasAnchor">
<a href="#tab-7-biomarker" class="anchor"></a>Tab 7: Biomarker</h3>
<div id="run-biomarker" class="section level5">
<h5 class="hasAnchor">
<a href="#run-biomarker" class="anchor"></a>Run Biomarker</h5>
<p>To identify biomarkers for a specific binary classification problem, users need to specify the taxonomy level and target variable. In the Advanced Options, users could also specify the number of CV repeats, number of CV folds, and top biomarker proportion. For example, with a 3-repeats 3-fold cross validation, animalcules will randomly split the dataset into 3 fold and run CV, then this procedure is repeated 3 times (each time has different random data split). The top biomarker proportion defines the threshold for selecting biomarkers: animalcules will generate an classification model based importance score for each microbe/feature, and will choose the top 20% (suppose the proportion is 0.2 as default) features as the biomarkers.</p>
<p>Also, users could choose binary classification models including logistic regression and random forest. After clicking the button “Run”, the biomarker list will shouw up at the right-hand side.</p>
<p>Note:</p>
<ul>
<li>If the dataset is too small or unbalanced, cross-validation can’t be applied. You will see error messages like: NA/NaN/Inf in foreign function call.</li>
<li>The target variable can not contain any special characters, otherwise there will be an error.</li>
</ul>
<p>Instructions:</p>
<ul>
<li>Select taxonomy level in the menu (default: genus).</li>
<li>Select the target variable for biomarker identification.</li>
<li>(Optional) Select number of CV folds (default: 3).</li>
<li>(Optional) Select number of CV repeats (default: 3).</li>
<li>(Optional) Select top biomarker proportion based on importance score (default: 0.2, representing 20%).</li>
<li>(Optional) Select model (default: logistic regression).</li>
<li>Click the button “Run”</li>
</ul>
<p>Running time:</p>
<ul>
<li>Test dataset with 30 samples and 427 microbes: 8.5s</li>
<li>Test dataset with 587 samples and 203 microbes: 32.4s</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_039.png"></p>
</div>
<div id="importance-plot" class="section level5">
<h5 class="hasAnchor">
<a href="#importance-plot" class="anchor"></a>Importance Plot</h5>
<p>Ranked feature importance score plot for the identified biomarkers is showed here. The higher score is, the more important this feature (species, genus, ..) is regarding the prediction power.</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_040.png"></p>
</div>
<div id="cv-roc-plot" class="section level5">
<h5 class="hasAnchor">
<a href="#cv-roc-plot" class="anchor"></a>CV ROC Plot</h5>
<p>The identified biomarkers were used to re-train the model via a cross-validation, and ROC plot is showed automatically in this subtab.</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_041.png"></p>
</div>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/animalcules">https://​www.bioconductor.org/​packages/​animalcules</a>
</li>
<li>Browse source code at <br><a href="https://github.com/compbiomed/animalcules">https://​github.com/​compbiomed/​animalcules</a>
</li>
<li>Report a bug at <br><a href="https://github.com/compbiomed/animalcules/issues">https://​github.com/​compbiomed/​animalcules/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Yue Zhao <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>

      <footer><div class="copyright">
  <p>Developed by Yue Zhao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
